core/src/main/java/hudson/security/TokenBasedRememberMeServices2.java

```diff
@@ -161,10 +161,12 @@ public void loginSuccess(HttpServletRequest request, HttpServletResponse respons
 
     @Override
     public Authentication autoLogin(HttpServletRequest request, HttpServletResponse response) {
-        if(Jenkins.get().isDisableRememberMe()){
+        // as this filter could be called during restart, this corrects at least the symptoms
+        Jenkins j = Jenkins.getInstanceOrNull();
+        if (j != null && j.isDisableRememberMe()) {
```

Comment:
It seems doubtful that without the `Jenkins` singleton, we have enough working that ending up in `retrieveAuthFromCookie` will ultimately be successful (`User#getById`?).

Just return `null` when `jenkins == null`, and return?

Reply:
Very good point, some implementation of the userDetailsService (like for the embedded SecurityRealm) are calling such mechanisms.